src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -102,8 +102,7 @@ sharded_store::make_valid_schema(canonical_schema schema) {
     case schema_type::avro:
         co_return make_avro_schema_definition(schema.def().raw()()).value();
     case schema_type::protobuf:
-        co_return co_await make_protobuf_schema_definition(
-          *this, std::move(schema));
+        co_return co_await make_protobuf_schema_definition(*this, schema);
```

Comment:
what's the undefined behavior this looks ok. unless I'm missing something, does this change fixing the issue point at the possibility that there is a different issue being masked?

Reply:
I can reliable create the crash:

* With the test cases.
* Manually running a curl 10 times.
* In the Materialize test suite.

This one change fixes them all.

Reply:
Typically what happens is that it appears as though the `schema` `type()` is lost, the variant inside `valid_schema_def` then assumes the first type, `avro`, then a `valid_protobuf` schema is effectively `reinterpret_cast` as `valid_avro`, and it all goes very wrong from there, usually deep in the memory allocator.

It's a Heisenbug though, it only occurs in release mode, and adding instrumentation often fixes it. It does reproduce in the debugger, but lots of stuff has been inlined, so it's hard to figure out what went wrong.